packages/aws_common/test/logging/aws_logger_test.dart

```diff
@@ -220,5 +222,76 @@ void main() {
       expect(loggerPlugin.timesCalled, 5);
       expect(globalLoggerPlugin.timesCalled, 0);
     });
+
+    test('Get parent returns parent in the logger hierarchy', () {
+      final logger = AWSLogger().createChild('child').createChild('grandChild');
+      expect(logger.parent, AWSLogger().createChild('child'));
+      expect(logger.parent?.parent, AWSLogger());
+      expect(logger.parent?.parent?.parent, isNull);
+    });
+
+    test('Get children returns children in the logger hierarchy', () {
+      final logger = AWSLogger();
+      final childLogger1 = logger.createChild('child-1');
+      final childLogger2 = logger.createChild('child-2');
+      final grandChildLogger = childLogger1.createChild('grandChild');
+
+      expect(logger.children.contains(childLogger1), isTrue);
+      expect(logger.children.contains(childLogger2), isTrue);
+      expect(childLogger1.children.contains(grandChildLogger), isTrue);
+      expect(grandChildLogger.children, isEmpty);
+    });
+
+    test('getPlugin returns registered plugin in the logger hierarchy', () {
+      final loggerPlugin = MockLoggerPlugin();
+      final logger = AWSLogger()..registerPlugin(loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+
+      final childLogger = logger.createChild('child');
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+
+      childLogger.unregisterPlugin(loggerPlugin);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+
+      logger.unregisterPlugin(loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), isNull);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), isNull);
+
+      childLogger.registerPlugin(loggerPlugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), isNull);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), loggerPlugin);
+    });
+
```

Comment:
Can you add a test for the following:

```dart
final logger = AWSLogger();
final childLogger = logger.createChild('child');
final plugin1 = MockLoggerPlugin();
final plugin2 = MockLoggerPlugin2();

// Test 1
logger.registerPlugin(plugin1);
childLogger.registerPlugin(plugin2);

expect(logger.getPlugin<MockLoggerPlugin>(), ...);
expect(logger.getPlugin<MockLoggerPlugin2>(), ...);
expect(childLogger.getPlugin<MockLoggerPlugin>(), ...);
expect(childLogger.getPlugin<MockLoggerPlugin2>(), ...);

// Test 2 (switch registration)
logger.registerPlugin(plugin2);
childLogger.registerPlugin(plugin1);

// same expectations
```

Reply:
Can you add this? I don't see any tests which will evaluate the same thing (`is T` vs. `runtimeType == T`)

Reply:
there is a test case below `registerPlugin allows registering plugins of same superclass` that covers the testing for (is T vs. runtimeType == T). 
what do we want to test with the above test case?

Reply:
If you change `runtimeType == T` to `is T` in your code all your current tests pass. The tests I'm proposing do not pass.

Reply:
changing the `runtimeType == T` to `is T` cause the `registerPlugin handles registering one plugin per type in the logger hierarchy` to fail. so this use case is covered.

Reply:
That test shouldn't fail when swapping the check. (nor does it fail for me - so I'm not sure why it would fail for you).

The test that should fail, if any, is the one you mentioned above: `registerPlugin allows registering plugins of same superclass`. However, this test passes when changing the type check.

Please double check. I cannot make tests fail by changing to `is T` and would recommend you add the tests mentioned to prevent a regression.

Reply:
`registerPlugin allows registering plugins of same superclass` did not fail with changing `== T` to `is T` because `MockLoggerPlugin2 extends MockLoggerPlugin` and registering plugin1 and then plugin2 does not make it fail. 
updated the test case to registere plugin2 first and then plugin1 and now it fails if changing `== T` to `is T`. also extended the test case to test registering to a logger and also registering to logger and its child.
now the test case fails if changing `== T` to `is T` or changing `== Plugin` to `is Plugin`.

Reply:
Yes, that's correct. Glad we're on the same page.